donate.html

```diff
@@ -946,6 +1044,52 @@ <h5 class="margin-bottom-normal">{% filter ak_text:"donate_billing_information"
     return !(results[0] || results[1] || results[2]);
   }
 
+  function checkout_step_track(step, payment) {
+      var amount_container = $('li.ak-amount-container label.ak-radio-checked');
+      var amount;
+      var price;
+      var frequency;
+      var currency = document.querySelector('li.ak-amount-container>label>span.ak-currency-sym').innerHTML;
+      var button_amount = amount_container.find('span.ak-amount-button-amount').text();
+      var custom_amount = document.querySelector('#ak-other-amount-field').value;
+      if (!custom_amount) {
+      amount = button_amount;
+      amount_type = 'preset';
+      price = button_amount;
+    } else {
+      amount = 'custom';
+      amount_type = 'custom';
+      price = custom_amount;
+    }
+      if ($('input[name="donation_type"][type="hidden"]').val() === 'recurring') {
+        frequency = 'recurring';
+      } else {
+        frequency = 'single';
+      }
+   var country = document.getElementById("country").value
+    dataLayer.push({ ecommerce: null });  // Clear the previous ecommerce object.

```

Comment:
Event documentation for reference:
https://developers.google.com/analytics/devguides/collection/ua/gtm/enhanced-ecommerce#checkoutstep
This function is currently being used to handle 2 checkout steps.